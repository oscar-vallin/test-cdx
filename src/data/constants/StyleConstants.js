export const StyleConstants = {
  DIRECTION_ROW: 'row',
  DIRECTION_COLUMN: 'column',

  SPINNER_XSMALL: 'xs',
  SPINNER_SMALL: 'sm',
  SPINNER_MEDIUM: 'md',
  SPINNER_LARGE: 'lg',
  SPINNER_XLARGE: 'xl',
};
