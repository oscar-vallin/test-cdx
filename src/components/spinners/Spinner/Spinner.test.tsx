import React from 'react';
import { shallow } from 'enzyme';
import { Spinner as Component } from './index.js';
import { StyleConstants } from '../../../data/constants/StyleConstants.js';
import { getSpinnerSize } from './Spinner.handlers.js';

describe('Spinner', () => {
  const tree = shallow(<Component size={StyleConstants.SPINNER_SMALL} label="loading ..." />);

  it('Should be defined', () => {
    expect(Component).toBeDefined();
  });

  it('Should render correctly', () => {
    expect(tree).toMatchSnapshot();
  });

  it('Should have a size', () => {
    expect(tree.props().size).toEqual(getSpinnerSize(StyleConstants.SPINNER_SMALL));
  });

  it('Should have a label', () => {
    expect(tree.props().label).toEqual('loading ...');
  });
});
