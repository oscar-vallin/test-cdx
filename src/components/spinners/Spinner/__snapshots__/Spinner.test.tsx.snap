// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Should render correctly 1`] = `ShallowWrapper {}`;
